xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
+  double totalLatency = DVD_SEC_TO_TIME(TotalLatency());
```

Comment:
> I would have thought by requesting a visual with double buffers we get two buffers.

Mesa-egl on Wayland does not use a fixed number of buffers (don't know about X). Every time you call eglSwapBuffers, it submits the buffer that was currently being rendered into to the compositor. The compositor uses it to draw the screen. When screen redraw is done and the buffer will not be used anymore, the compositor tells mesa so that it can use the buffer for drawing again (if mesa would reuse the buffer before that it would lead to tearing). Mesa will allocate and cycle through as many buffers as needed such that drawing can happen (after eglSwapBuffers) on a buffer that is not in use by the compositor.

> We need the time from last vblank. If swapbuffers blocks, then after. On X11 it is not that trivial, because swapBuffers does not necessarily block (see above and X11 on my branch)

On Wayland, eglSwapBuffers also does not necessarily block (no blocking if we are late and missed the last frame hint), but also if it does block it does not follow vblank timing. swapBuffers is synchronized to vblank in the sense that if you render fast enough and have an output with 60 Hz, swapBuffers will block so that you render 60 fps on average. But the exact point in time swapBuffers returns is usually somewhere in the middle between two vblanks, when the compositor decides that it would be a good time for drawing.